### PR TITLE
[Solved] - Do not trigger transactions on the wrong network

### LINKED ISSUE #100: Do not trigger transactions on the wrong network
If the user logs in on one network, opens a publication, then switches networks and proceeds to initiate a transaction (say to make a post or modify some permissions), the app will trigger the transaction on the wrong network. This costs the user money and doesn't affect the intended change.

If the user switches networks to something other than the network that the currently selected publication lives on, we SHOULD prompt them/their wallet to switch back to the correct network before initiating the transaction.

We MUST never initiate a transaction on the wrong network.

### PR DESCRIPTION
**Pull request by juliopavila**
closed #100 

Record:

https://user-images.githubusercontent.com/19823989/175619496-a8613adc-e677-4766-bcec-79d99e95aea4.mov

### CODE REVIEW

**Comment by auryn-macmillan** [comment]
It would be good if the toast mentioned which network to switch to. Something like `Wrong network. Please switch to {network name}.`.

**Review by manboy-eth — CHANGES_REQUESTED** [review]
In my view, the cleanest solution to this is: always disconnecting the wallet when the user changes the network.

This is already implemented. However, there is a bug that makes the wallet not disconnect when we change to `Mainnet`.
I think fixing this bug should solve the #100.

Since it is not possible to connect a wallet to the "wrong network", it should not be possible to reach a state where a transaction can be executed on the wrong network.

File: packages/app/src/utils/validation.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { SupportedChain, SupportedChainId } from "../constants/chain"
+
+export const checkIsValidChain = (currentChain: number): boolean => {
```
Comment (manboy-eth) [inline packages/app/src/utils/validation.ts]: If we are not in a publication, it should return `true` as long as we are connected to one of the supported chains.

If not, it fails when creating a new publication,

**Comment by manboy-eth** [comment]
> In my view, the cleanest solution to this is: always disconnecting the wallet when the user changes the network.
> 
> This is already implemented. However, there is a bug that makes the wallet not disconnect when we change to `Mainnet`. I think fixing this bug should solve the #100.
> 
> Since it is not possible to connect a wallet to the "wrong network", it should not be possible to reach a state where a transaction can be executed on the wrong network.

I was wrong in my assumption, sorry. It seems that as long as you switch to a wallet that has already been connected to Tabula, it will not disconnect the wallet.

I still think disconnecting the wallet whenever the network changes might be the best solution.
Since there are already checks for the network/chain when connecting a wallet, it might solve the problem (instead of having two independent "chain validation solutions"). See: https://github.com/gnosis/tabula/blob/2a690668c1210b30e6554de58cd6053e6f6d4757/packages/app/src/components/views/wallet/WalletView.tsx#L64

Perhaps there is another reason we want users to be able to change between networks without disconnecting the wallet?

**Review by manboy-eth — APPROVED** [review]
LGTM

**Pull request merged**